### PR TITLE
Ftr : nacos config

### PR DESCRIPTION
**Pull request by zouyx**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by AlexStocks** [comment]
zouyx,  u'd better add some remark for those open funcs.

**Comment by zouyx** [comment]
i will fix it today , fixed test case yesterday

File: config_center/nacos/client.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+// WithNacosName Set nacos name
+func WithNacosName(name string) option {
+	return func(opt *options) {
+		opt.nacosName = name
+	}
+}
+
+// ValidateNacosClient Validate nacos client , if null then create it
+func ValidateNacosClient(container nacosClientFacade, opts ...option) error {
```
Comment (flycash) [inline config_center/nacos/client.go]: Is possible that this method is invoked concurrently? If so, if-null and then create will cause some problem.

File: config_center/nacos/client.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+			close(n.exit)
+		},
+	}
+
+	svrConfList := []nacosconst.ServerConfig{}
+	for _, nacosAddr := range n.NacosAddrs {
+		split := strings.Split(nacosAddr, ":")
+		port, err := strconv.ParseUint(split[1], 10, 64)
+		if err != nil {
+			continue
```
Comment (flycash) [inline config_center/nacos/client.go]: log the error

File: config_center/nacos/factory.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/config_center"
+	"github.com/apache/dubbo-go/config_center/parser"
+)
+
+func init() {
+	extension.SetConfigCenterFactory("nacos", func() config_center.DynamicConfigurationFactory { return &nacosDynamicConfigurationFactory{} })
```
Comment (flycash) [inline config_center/nacos/factory.go]: I think the nacosDynamicConfigurationFactory is singleton but the `func()` will always create new one.

File: config_center/nacos/impl.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		return false
+	default:
+		return true
+	}
+}
+
+func (r *nacosDynamicConfiguration) closeConfigs() {
+	r.cltLock.Lock()
+	defer r.cltLock.Unlock()
+	logger.Infof("begin to close provider nacos client")
```
Comment (flycash) [inline config_center/nacos/impl.go]: How about change to:
``` go
r.cltLock.Lock()
client := r.client
r.client = nil
r.cltLock.Unlock()
client.Close()
```

File: config_center/nacos/listener.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		newListener[listener] = cancel
+		l.keyListeners.Store(key, newListener)
+	} else {
+		// TODO check goroutine alive, but this version of go_nacos_sdk is not support.
+		logger.Infof("profile:%s. this profile is already listening", key)
+	}
+}
+
+func (l *nacosDynamicConfiguration) removeListener(key string, listener config_center.ConfigurationListener) {
+	// TODO not supported in current go_nacos_sdk version
```
Comment (flycash) [inline config_center/nacos/listener.go]: log some information...

File: config_center/nacos/factory.go (hunk 3 above)
Comment (zouyx) [reply config_center/nacos/factory.go]: you can check zookeeper implement, it is the same to nacos.
https://github.com/apache/dubbo-go/blob/33f82cd154a3ee5ef861c6465333c4945c9d4b1a/config_center/zookeeper/factory.go#L28

File: config_center/nacos/client.go (hunk 1 above)
Comment (zouyx) [reply config_center/nacos/client.go]: may be invoked concurrently, so it has a lock in ``` container.SetNacosClient(newClient) ``` and ``` container.NacosClient().SetClient(&client) ```.  Finally added if-null logic .

**Comment by AlexStocks** [comment]
LGTM

**Pull request merged**